plugins/woocommerce-blocks/assets/js/blocks/attribute-filter/frontend.ts

```diff
@@ -17,7 +17,8 @@ const getProps = ( el: HTMLElement ) => {
 };
 
 renderFrontend( {
-	selector: '.wp-block-woocommerce-attribute-filter',
+	selector:
+		'div:not(.wp-block-woocommerce-filter-wrapper) > .wp-block-woocommerce-attribute-filter',
```

Comment:
Thanks for the quick fix! @dinhtungdu 

 If I'm not mistaken, it looks like `.wp-block-woocommerce-attribute-filter` is not directly under `.wp-block-woocommerce-filter-wrapper`, there is a `.wc-blocks-filter-wrapper div` in between.  

Should we update the selector to 
`.wp-block-woocommerce-attribute-filter:not(.wp-block-woocommerce-filter-wrapper .wp-block-woocommerce-attribute-filter)`?


<img width="1147" alt="Screenshot 2024-10-02 at 13 40 37" src="https://github.com/user-attachments/assets/01f0c2ca-466f-4d43-8b85-55e90ac0cf5d">

Reply:
That div is rendered by React, it does not exist in the original DOM. We need to target the DOM on rendering.

Reply:
Thank you for the clarification! 

In that case, if the filter wrapper renders the attribute filter block first and alters the DOM structure, then the attribute filter block itself renders, will the selector still function correctly?

My understanding is that the attribute filter selector will be a match since it would already be present in the DOM at that point. Do I miss anything?

Reply:
I tried to simulate the scenario by delaying the rendering of the attribute filter block. The issue seems to still happen. 🤔 

```js
setTimeout( () => {
	renderFrontend( {
		selector:
			'div:not(.wp-block-woocommerce-filter-wrapper) > .wp-block-woocommerce-attribute-filter',
		Block,
		getProps,
	} );
}, 3000 );
```

Reply:
That's a good point! Let me test it with your timeout trick to see what happens.

Reply:
Perhaps we can be more aggressive here, check if there is `.wp-block-woocommerce-filter-wrapper` in the DOM, then we bail the rendering for individual filter blocks, based on the assumption that if users upgrade filter blocks, they upgraded all blocks already. What do you think @chihsuan ?

Reply:
I don't want to remove the extra div (even though I consider it's not necessary) again for the backward compat reason.

Reply:
>  based on the assumption that if users upgrade filter blocks, they upgraded all blocks already. 

If the assumption is correct, then it should work. If users only partially upgrade the blocks, then the individual filters will not be rendered. 😅 I'm not sure if it's likely to happen, do users manually upgrade the blocks one by one?

Reply:
Hmm, let me try another CSS query.

Reply:
@chihsuan I think I got it, can you please test the latest commit.

Reply:
Test with timeout and it's working.

Reply:
Thanks for the update! I've tested the latest commit and it works as expected. 

I tested the following scenarios:

1. Shop page with new attribute filters, price filter, stock filter, rating filter, and active filters.
2. Shop page mixed with old and new attribute filters.
3. Shop page with only old attribute filters.
4. With and without the delay in rendering the attribute filter block.
5. With and without network cache.

I think we're good to go! 🙌